mer/utils.py

```diff
@@ -8,45 +8,79 @@
 GAP = "***"
 
 
+def convert_err_str_to_penalty(err_str):
+    """
+    param: err_str: string of error codes (m = minor, s = standard, e = serious)
+    returns: float representing total penalty (m = 0.25, s = 0.5, e = 1.0)
+    """
+    err_map = {"m": 0.25, "s": 0.5, "e": 1.0}
+
+    penalty = sum([err_map[char] for char in err_str])
+    return penalty
+
+
+# def convert_err_str_to_dict(err_str):
+#     """
+#     param: err_str: string of error codes (m = minor, s = standard, e = serious)
+#     returns: readable dict of errors
+#     """
+#     return {'minor': err_str.count("m"),
+#             'standard': err_str.count("s"),
+#             'serious': err_str.count("e")}
+
+
+def convert_err_str_to_dict(err_str):
+    """
+    param: err_str: string of error codes (m = minor, s = standard, e = serious)
+    returns: readable dict of errors
+    """
+    err_map = {"m": "minor", "s": "standard", "e": "serious"}
+    full_err_str = [err_map[err] for err in err_str]
+    return " ".join(full_err_str)
+
+
 def majority_voting(continuations, prompt):
     # Loop over text in continuations and extract results
     predictions = []
-    penalities = []
+    err_strings = []
     for text in continuations:
-        error_counts_dict, penalty = prompt.get_result(text)
-        penalities.append(penalty)
+        error_counts_dict, err_str = prompt.get_result(text)
+        err_strings.append(err_str)
         predictions.append(error_counts_dict)
 
     # Run majority voting given the predicted error tpes
-    counts = Counter(penalities)
-    voted_penality, vote_count = counts.most_common()[0]
+    counts = Counter(err_strings)
+    voted_err, vote_count = counts.most_common()[0]
+    if voted_err is not None:
+        penalty = convert_err_str_to_penalty(voted_err)
+        voted_err = convert_err_str_to_dict(voted_err)
+    else:
+        penalty = None
 
-    result = {
-        "predictions": predictions,
-        "voted_penality": voted_penality,
-        "vote_count": vote_count,
-    }
+    result = {"predictions": predictions, "voted_penalty": penalty, "voted_error": voted_err, "vote_count": vote_count}
 
-    return voted_penality, result
+    return penalty, result
 
 
 def get_alignment(ref_text, rec_text):
```

Comment:
add `remove_punc=True` and add if statement below rather than leaving it commented out

Reply:
yeah default should be what you believe leads to the most robust performance. Don't bother with user-facing

Reply:
done